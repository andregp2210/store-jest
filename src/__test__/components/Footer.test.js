import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('<Footer />', () => {
  const footer = mount(<Footer />);
  test('Component renders', () => {
    expect(footer.length).toEqual(1);
  });
  test('Component renders title', () => {
    expect(footer.find('[data-testid="footerTitle"]').text()).toEqual(
      'Platzi Stores'
    );
  });
  test('Component renders rights', () => {
    expect(footer.find('.Footer-copy').text()).toEqual(
      'Todos los Izquierdos Reservadoss'
    );
  });
});

describe('Footer Snapshot', () => {
  test('Footer UI Snapshot', () => {
    const footer = create(<Footer />);
    expect(footer.toJSON()).toMatchSnapshot();
  });
});
