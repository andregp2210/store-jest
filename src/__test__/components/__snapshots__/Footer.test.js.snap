// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Footer UI Snapshot 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
    data-testid="footerTitle"
  >
    Platzi Store
  </p>
  <p
    className="Footer-copy"
  >
    Todos los Izquierdos Reservados
  </p>
</div>
`;
